includes/apple-exporter/components/class-table.php

```diff
@@ -98,212 +141,115 @@ public function register_specs() {
+					'horizontalAlignment' => '#table_body_horizontal_alignment#',
+					'padding'             => '#table_body_padding#',
+					'textStyle'           => [
+						'fontName'   => '#table_body_font#',
+						'fontSize'   => '#table_body_size#',
+						'lineHeight' => '#table_body_line_height#',
+						'textColor'  => '#table_body_color#',
+						'tracking'   => '#table_body_tracking#',
 					],
-					'conditions' => [
-						'minSpecVersion'       => '1.14',
-						'preferredColorScheme' => 'dark',
+					'verticalAlignment'   => '#table_body_vertical_alignment#',
+				],
+				'columns'     => [
+					'divider' => [
+						'color' => '#table_border_color#',
+						'style' => '#table_border_style#',
+						'width' => '#table_border_width#',
 					],
 				],
-			];
-			$table_row_col_base_conditional = [
-				[
-					'selectors'  => [
-						[ 'even' => true ],
-						[ 'odd' => true ],
+				'headerCells' => [
+					'backgroundColor'     => '#table_header_background_color#',
+					'horizontalAlignment' => '#table_header_horizontal_alignment#',
+					'padding'             => '#table_header_padding#',
+					'textStyle'           => [
+						'fontName'   => '#table_header_font#',
+						'fontSize'   => '#table_header_size#',
+						'lineHeight' => '#table_header_line_height#',
+						'textColor'  => '#table_header_color#',
+						'tracking'   => '#table_header_tracking#',
 					],
-					'conditions' => [
-						'minSpecVersion'       => '1.14',
-						'preferredColorScheme' => 'dark',
+					'verticalAlignment'   => '#table_header_vertical_alignment#',
+				],
+				'headerRows'  => [
+					'divider' => [
+						'color' => '#table_border_color#',
+						'style' => '#table_border_style#',
+						'width' => '#table_border_width#',
 					],
 				],
-			];
-		}
+				'rows'        => [
+					'divider' => [
+						'color' => '#table_border_color#',
+						'style' => '#table_border_style#',
+						'width' => '#table_border_width#',
+					],
+				],
+			],
+		];
 
-		// The following block sets:
-		// Dark Background Color of Cells
-		// Dark Text Color of Cells.
-		$dark_bg_text_conditional = [];
-		if (
-			! empty( $table_body_background_color_dark ) ||
-			! empty( $table_body_color_dark )
-		) {
-			$dark_bg_text_conditional = [
-				'conditional' => [ $table_cell_base_conditional[0] ],
-			];
-		}
+		$this->register_spec(
+			'default-table',
+			__( 'Table Style', 'apple-news' ),
+			$default_table_styles
+		);
 
-		if ( ! empty( $table_body_background_color_dark ) ) {
-			$dark_bg_text_conditional['conditional'][0]['backgroundColor'] = '#table_body_background_color_dark#';
-		}
+		if ( $this->dark_table_colors_exist ) {
 
-		if ( ! empty( $table_body_color_dark ) ) {
-			$dark_bg_text_conditional['conditional'][0]['textStyle'] = [
-				'textColor' => '#table_body_color_dark#',
-			];
-		}
+			// Start with default-table styles as a base.
+			// Then modify dark styles where applicable.
+			$dark_table_styles = $default_table_styles;
```

Comment:
`git` made a mess parsing the diff in this file.  The crux of the work is here.

`$default_table_styles` is declared as normal, then `$dark_table_styles` is copied and updated with dark mode styles (if they're set in the theme).  Then the two assoc arrays are registered as `default-table` and `dark-table`.